### DISCUSSION TITLE
window rows and cols

### DISCUSSION THREAD

**Discussion by futsuuu** [wezterm/wezterm#2841] [Q&A] [answered]
I know how to get the pixel size of the window and the columns/rows of the pane, but I don't know how to get the columns/rows of the window. 
If a method already exists, Please let me know how to do this.

**Comment by wez** [comment]
What exactly are you trying to do?

**Comment by futsuuu** [comment]
I want to do [this](https://github.com/wez/wezterm/issues/1914) with non fancy tab bar style.

**Comment by wez** [comment, marked as answer]
There isn't a function that returns exactly that metric, but you can derive it for yourself:

Assuming that you are doing this from inside `format-tab-title`:

The `panes` parameter has information about each pane.

If you compute `pane.left + pane.width` you have the *x* position of the right hand side of that pane.
Compute the maximum of that across all of the `panes` and you have the width of the terminal in cells.

**Comment by futsuuu** [comment]
I hadn't thought of that idea! Thanks😎
